ch_tools/chadmin/cli/partition_group.py

```diff
@@ -119,20 +120,38 @@ def list_partitions_command(ctx, **kwargs):
     help="Enable dry run mode and do not perform any modifying actions.",
 )
 @pass_context
-def attach_partitions_command(ctx, dry_run, all_, database, table, **kwargs):
+def attach_partitions_command(
+    ctx, all_, database, table, partition_id, keep_going, dry_run, **kwargs
+):
     """Attach one or several partitions."""
-    if not any((all_, database, table)):
+    if not any((all_, database, table, partition_id)):
         ctx.fail(
             "At least one of --all, --database, --table, --partition options must be specified."
         )
 
     partitions = get_partitions(
-        ctx, database, table, detached=True, format_="JSON", **kwargs
+        ctx,
+        database,
+        table,
+        partition_id=partition_id,
+        detached=True,
+        format_="JSON",
+        **kwargs,
     )["data"]
     for p in partitions:
-        attach_partition(
-            ctx, p["database"], p["table"], p["partition_id"], dry_run=dry_run
-        )
+        try:
+            attach_partition(
+                ctx,
+                database=p["database"],
+                table=p["table"],
+                partition_id=p["partition_id"],
+                dry_run=dry_run,
+            )
+        except Exception as e:
+            if keep_going:
+                print(repr(e))
```

Comment:
Why not logging ?

Reply:
To not hide error from a user and output it to stdout.

Here's an example output: 

```
# chadmin partition attach --all -k
ALTER TABLE `db1`.`events` ATTACH PARTITION ID '20230306'

ClickhouseError("Code: 499. DB::Exception: The specified key does not exist.: while reading key: cloud_storage/cluster1001/shard1/r0000000000000000000000000000000000000000000011110011101111111001-file-ojvazctfenasiyjtsvzzsjmmubpdmgft, from bucket: cloud-storage-cluster1001. (S3_ERROR) (version 23.8.6.16 (official build))\n\nQuery: ALTER TABLE `db1`.`events` ATTACH PARTITION ID '20230306'",)

ALTER TABLE `db1`.`events` ATTACH PARTITION ID '20230326'
```

We can add logging in addition to outputting error to stdout. And error reporting to stdount not as good as it can be. But for now I implemented `--keep-going` option in consistent way with other commands that support it.